### DISCUSSION TITLE
Add ability to check checkbox and toggle with `data-checked` or `data-selected` property

### DISCUSSION THREAD

**Discussion by apstanisic** [saadeghi/daisyui#2639] [Ideas / feature requests]
Not every UI framework is using `<input type="checkbox" class="toggle" checked />` for checkbox and toggle.

[`react-aria-components`](https://react-spectrum.adobe.com/react-aria/Checkbox.html) renders hidden input and we only render UI with divs.
We can set `<div className="toggle" aria-checked={isSelected} />`, but that might mess up with a11y.

I propose that additional selector is added for checked, `.toggle[aria-checked="true"]` or `.toggle[aria-selected="true"]`.

I can create PR for this feature.

**Comment by saadeghi** [comment]
daisyUI already has that. Here's an example of using a div with `aria-checked`:  
https://play.tailwindcss.com/lRpWXVI36i

**Comment by apstanisic** [comment]
Hey @saadeghi , sorry, I made a typo. I meant to write `data-checked` instead of `aria-checked`. As temporary workaround I'm currently using `aria-checked`. I do not know if having 2 checked state elements is a11y correct?
Here is an checkbox example. `react-aria-components` renders invisible input with `checked` property, plus has additional div with `aria-checked` 

```html
<label class="my-1" data-rac="" data-selected="true">
    <!-- library renders invisible span that contains input, which contains checked property -->
    <span style="border: 0px; clip: rect(0px, 0px, 0px, 0px); clip-path: inset(50%); height: 1px; margin: -1px; overflow: hidden; padding: 0px; position: absolute; width: 1px; white-space: nowrap;">
        <input name="test" type="checkbox" checked="">
    </span>
    <div class="flex gap-x-2 items-center">
 <!-- We are using additional checked property with `aria-checked` -->
        <div class="du-checkbox du-checkbox-md bg-base-100" aria-checked="true"></div>
    </div>
</label>
```

**Reply by saadeghi** [reply]
>I do not know if having 2 checked state elements is a11y correct?

`data-checked` is meaningless to screen readers. They understand native HTML checkbox or `role="checkbox" aria-checked=true`
Anything else is not a checkbox and not accessible to screen readers. I don't have any plans to add styles for `data-checked` or `data-selected` because those are not the correct attributes to show the checkbox state. It should be either a HTML checkbox or `role="checkbox" aria-checked=true`
